tests with `:` in the description string generate invalid methods

This test definition crashes minitest because the macro output isn't a valid function signature:

```crystal
  it "when attr: false" do
    # blah
  end
```

 > 1 |      def test_when_attr:_false
Error: a space is mandatory between ':' and return type
make: *** [test] Error 1

I didn't dig into the test suite to sort out _why_ `:` is included in the [regex whitelist](https://github.com/ysbaddaden/minitest.cr/blob/main/src/spec.cr#L43), but there it is.

--- a/src/spec.cr
+++ b/src/spec.cr
@@ -25,10 +25,9 @@ module Minitest
     macro describe(name, &block)
       {%
         class_name = name.id.stringify
-          .gsub(/[^0-9a-zA-Z:]+/, "_")
-          .gsub(/^_|_$/, "")
-          .split("_").map { |s| [s[0...1].upcase, s[1..-1]].join("") }.join("")
-          .split("::").map { |s| [s[0...1].upcase, s[1..-1]].join("") }.join("::")
+          .split("::")
+          .map(&.gsub(/[^0-9a-zA-Z]+/, "_").gsub(/^_|_$/, "").capitalize)
+          .join("::")
       %}
       class {{ class_name.id }}Spec < {{ @type }}
         def self.name
@@ -40,7 +39,7 @@ module Minitest
     end
 
     macro it(name = "anonymous", &block)
-      def test_{{ name.strip.gsub(/[^0-9a-zA-Z:]+/, "_").id }}
+      def test_{{ name.strip.gsub(/[^0-9a-zA-Z]+/, "_").id }}
         {{ yield }}
       end
     end
@@ -55,10 +54,9 @@ end
 macro describe(name, &block)
   {%
     class_name = name.id.stringify
-      .gsub(/[^0-9a-zA-Z:]+/, "_")
-      .gsub(/^_|_$/, "")
-      .split("_").map { |s| [s[0...1].upcase, s[1..-1]].join("") }.join("")
-      .split("::").map { |s| [s[0...1].upcase, s[1..-1]].join("") }.join("::")
+      .split("::")
+      .map(&.gsub(/[^0-9a-zA-Z]+/, "_").gsub(/^_|_$/, "").capitalize)
+      .join("::")
   %}
   class {{ class_name.id }}Spec < Minitest::Spec
     def self.name

--- a/test/spec_test.cr
+++ b/test/spec_test.cr
@@ -43,6 +43,10 @@ describe Minitest::Spec do
   describe("ending with special chars #") {}
   it(". has leading and ending special chars .") {}
 
+  describe("has ::colons:") do
+    it("has ::colons:") {}
+  end
+
   describe "let" do
     let(:foo) { Foo.new }
 

